src/signing_key/key.rs

```diff
@@ -72,22 +75,25 @@ impl TryFrom<&SigningKey> for secp256k1::SigningKey {
     type Error = ChainError;
     fn try_from(signer: &SigningKey) -> Result<Self, Self::Error> {
         match &signer.key {
-            Key::Mnemonic(phrase) => mnemonic_to_signing_key(phrase),
+            Key::Mnemonic(phrase) => mnemonic_to_signing_key(phrase, &signer.derivation_path),
 
             #[cfg(feature = "os_keyring")]
             Key::Keyring(params) => {
                 let entry = Entry::new(&params.service, &params.key_name);
-                mnemonic_to_signing_key(&entry.get_password()?)
+                mnemonic_to_signing_key(&entry.get_password()?, &signer.derivation_path)
             }
         }
     }
 }
 
-fn mnemonic_to_signing_key(mnemonic: &str) -> Result<secp256k1::SigningKey, ChainError> {
+fn mnemonic_to_signing_key(
+    mnemonic: &str,
+    derivation_path: &str,
+) -> Result<secp256k1::SigningKey, ChainError> {
     let seed = bip32::Mnemonic::new(mnemonic, bip32::Language::English)
         .map_err(|_| ChainError::Mnemonic)?
         .to_seed("");
 
-    secp256k1::SigningKey::derive_from_path(seed, &DERVIATION_PATH.parse().unwrap())
+    secp256k1::SigningKey::derive_from_path(seed, &derivation_path.parse().unwrap())
```

Comment:
We should stop unwrapping here and throw a `ChainError::DerviationPath` error: `&derivation_path.parse().unwrap()`

I was only unwrapping because `derivation_path` was always the same value before.

Reply:
Agreed, most recent commit fixes that.